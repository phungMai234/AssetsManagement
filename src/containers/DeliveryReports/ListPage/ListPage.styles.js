import styled from 'styled-components';

const Wrapper = styled.div`
  table {
    th {
      text-align: center;
    }
    td {
      text-align: center;
    }
    .td-manager {
      width: 15%;
      word-wrap: break-word;
    }
  }
  margin: 5px;
  .row {
    margin-bottom: 20px;
  }
  .border-place {
    border-left: 1px solid #bfbfbf;
    display: flex;
    align-items: center;
    button {
      margin-right: 10px;
      i {
        padding-right: 5px;
      }
    }
  }

  .row-button {
    display: flex;
    align-items: center;

    .btn-add {
      display: flex;
      justify-content: center;
      align-items: center;
      margin-right: 20px;

      span {
        margin-left: 5px;
      }
    }

    .btn-import {
      display: flex;
      justify-content: center;
      align-items: center;

      span {
        margin-left: 5px;
      }
    }
  }

  .input-select {
    background-color: #fff;
    border-radius: 3px;
    padding: 6px;
    width: 200px;
  }

  .file-text {
    stroke: #2196f3;
    width: 20px;
    height: 20px;
  }
`;
export default Wrapper;
