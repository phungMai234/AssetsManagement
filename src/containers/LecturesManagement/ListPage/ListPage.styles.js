import styled from 'styled-components';

const Wrapper = styled.div`
  .header-content {
    display: flex;
    align-items: center;
  }
  .btn-import {
    margin-left: 20px;
  }
  table {
    /* table-layout: fixed;
    .index {
      width: 5%;
    }
    .name {
      width: 20%;
    }
    .manager {
      width: 20%;
    }
    .note {
      width: 20%;
    }
    .edit_row,
    .delete_row {
      width: 5%;
    } */

    .no-result {
      text-align: center;
    }
  }
  td {
    vertical-align: middle;
  }
  .td-note {
    div {
      display: -webkit-box;
      max-width: 400px;
      -webkit-line-clamp: 3;
      -webkit-box-orient: vertical;
      text-overflow: ellipsis;
      overflow: hidden;
    }
  }
`;
export default Wrapper;
