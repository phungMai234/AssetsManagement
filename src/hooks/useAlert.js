import { useContext } from 'react';

import MessageContext from '../contexts/message';

const useAlert = /* istanbul ignore next */ () => {
  const context = useContext(MessageContext);

  if (context === undefined) {
    return;
  }
  return context;
};

export default useAlert;
