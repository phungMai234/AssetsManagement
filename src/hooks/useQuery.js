import { useState, useEffect, useCallback } from 'react';

import db from '../database';

export const useQuery = ({ url, ignoreCancel = false }) => {
  const [data, setData] = useState(null);
  const [loading, setLoading] = useState(false);
  const [errors, setErrors] = useState(null);
  const [forceRequest, setForceRequest] = useState(false);

  useEffect(() => {
    setLoading(true);

    db.collection(url).onSnapshot((querySnapshot) => {
      let result = [];
      querySnapshot.forEach((doc) => {
        result = [...result, { id: doc.id, ...doc.data() }];
      });
      setData(result);
      setErrors(null);
      setLoading(false);
    });
  }, [url, forceRequest, ignoreCancel]);

  const force = useCallback(() => {
    setForceRequest({});
  }, []);

  return { loading, data, errors, force };
};
