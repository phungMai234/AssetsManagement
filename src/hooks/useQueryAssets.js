import { useState, useEffect, useCallback } from 'react';

import db from 'database';

export const useQueryAssets = ({ ignoreCancel = false }) => {
  const [data, setData] = useState([]);
  const [loading, setLoading] = useState(false);
  const [errors, setErrors] = useState(null);
  const [forceRequest, setForceRequest] = useState(false);

  useEffect(() => {
    setLoading(true);

    db.collection('assets')
      .orderBy('model_number', 'asc')
      .onSnapshot((querySnapshot) => {
        let result = [];
        querySnapshot.forEach((doc) => {
          result.push({ id: doc.id, ...doc.data() });
        });
        setData(result);
        setErrors(null);
        setLoading(false);
      });
  }, [forceRequest, ignoreCancel]);

  const force = useCallback(() => {
    setForceRequest({});
  }, []);

  return { loading, data, errors, force };
};
