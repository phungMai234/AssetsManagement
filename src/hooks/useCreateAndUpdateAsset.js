import { useCallback } from 'react';
import { useHistory } from 'react-router-dom';

import { db, firebase } from 'database';
import useAlert from 'hooks/useAlert';
import focusOnTop from 'utils/focusOnTop';

const useCreateAndUpdateAsset = ({ data }) => {
  const { setAlert } = useAlert();
  const history = useHistory();

  const update = useCallback(
    (values, actions) => {
      actions.setSubmitting(true);

      const cloneValues = { ...values };

      const formatValues = {
        ...cloneValues,
        purchase_date: firebase.firestore.Timestamp.fromDate(values.purchase_date),
      };

      // check exist

      const formatValuesCloned = {
        image_detail: formatValues.image_detail,
        name: formatValues.name,
        description: formatValues.description,
      };

      // const currentSerialNumber = cloneValues.serial_number;
      // db.collection('assets')
      //   .where('serial_number', '==', currentSerialNumber)
      //   .get()
      //   .then((querySnapshot) => {
      //     querySnapshot.forEach((doc) => {
      //       if (doc.exists) {
      //         actions.setSubmitting(false);
      //         history.replace(`/dashboard/assets`);
      //         setAlert({ status: 'danger', message: 'Tài sản này đã tồn tại!' });
      //         focusOnTop();
      //         return;
      //       }
      //     });
      //   })
      //   .catch(() => {
      //     setAlert({ status: 'danger', message: 'Đã xảy ra lỗi hệ thống. Vui lòng thử lại!' });
      //   });

      //update edit
      if (data) {
        db.collection('assets')
          .doc(data?.id)
          .update({ ...formatValues })
          .then(() => {
            db.collection('assets')
              .where('model_number', '==', formatValues.model_number)
              .get()
              .then((querySnapshot) => {
                querySnapshot.forEach((doc) =>
                  db
                    .collection('assets')
                    .doc(doc?.id)
                    .update({ ...formatValuesCloned }),
                );
              });
            history.push(`/dashboard/assets`);
            setAlert({ status: 'success', message: 'Cập nhật thông tin thành công' });
          })
          .catch(() => {
            setAlert({ status: 'danger', message: 'Đã xảy ra lỗi hệ thống. Vui lòng thử lại!' });
          })
          .finally(() => {
            actions.setSubmitting(false);
            focusOnTop();
          });
        return;
      }
      // create new

      db.collection('assets')
        .add({ ...formatValues })
        .then(() => {
          history.push(`/dashboard/assets`);
          setAlert({ status: 'success', message: 'Tạo mới thành công' });
        })
        .catch(() => {
          setAlert({ status: 'danger', message: 'Đã xảy ra lỗi hệ thống. Vui lòng thử lại!' });
        })
        .finally(() => {
          actions.setSubmitting(false);
          focusOnTop();
        });
    },
    [data, history, setAlert],
  );

  return [update];
};

export default useCreateAndUpdateAsset;
