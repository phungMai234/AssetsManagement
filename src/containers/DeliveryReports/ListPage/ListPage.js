import React, { useState, useMemo } from 'react';

import { filter, includes, lowerCase } from 'lodash';
import { format } from 'date-fns';
import { Row, Col, Button, Form } from 'react-bootstrap';
import { useHistory } from 'react-router-dom';
import { Plus, FilePlus, FileText } from 'react-feather';

import Wrapper from './ListPage.styles';
import TablePaginationData from 'components/TablePaginationData/TablePaginationData';
import BoxSearch from 'components/BoxSearch/BoxSearch';
import { formatDateToString } from 'utils/helper';
import { useQuery } from 'hooks/useQuery';
import DatePickerInput from 'components/DatePickerInput';
import StatusBorrow from 'components/StatusBorrow';
import { LIST_STATUS } from 'utils/constant';

const ListPage = () => {
  const history = useHistory();

  const [params, setParams] = useState({});

  const { data: dataItems, loading: loadingItems } = useQuery({ url: 'orders' });

  const recordItems = useMemo(() => {
    const newData = (dataItems || []).map((record) => ({
      ...record,
      date_borrowed: formatDateToString(record?.date_borrowed?.seconds),
      date_return: formatDateToString(record?.date_return?.seconds),
    }));

    const filterParams = {
      status: params.status,
      date_borrowed: params.date_borrowed && format(params.date_borrowed, 'dd/MM/yyyy'),
    };

    !params.status && delete filterParams.status;
    !params.date_borrowed && delete filterParams.date_borrowed;

    const customData = params.keyword
      ? filter(newData, (item) => includes(lowerCase(item.user_name), lowerCase(params.keyword)))
      : newData;

    return filter(customData, filterParams);
  }, [dataItems, params.date_borrowed, params.keyword, params.status]);

  const restructureData = useMemo(() => {
    if (!recordItems) return [];
    return recordItems.map((record, index) => ({
      ...record,
      index: index + 1,
      status: <StatusBorrow status={record?.status} />,
      report_file: record?.report_file ? <FileText className="file-text" /> : '-',
      onClick: () => history.push(`/dashboard/delivery_reports/${record.id}/detail`),
    }));
  }, [recordItems, history]);

  return (
    <Wrapper>
      <Row>
        <Col md={3}>
          <BoxSearch value={params.keyword} onChange={(e) => setParams({ ...params, keyword: e.target.value })} />
        </Col>
        <Col md={3}>
          <DatePickerInput
            value={params.date_borrowed}
            onSelect={(date) => setParams({ ...params, date_borrowed: date })}
          />
        </Col>
        <Col md={3}>
          <Form.Control
            as="select"
            size="sm"
            value={params.status || ''}
            onChange={(e) => setParams({ ...params, status: e.target.value })}
          >
            <option key="" value="">
              Chọn tất cả
            </option>

            {(LIST_STATUS || []).map((status) => (
              <option key={status.value} value={status.value}>
                {status.label}
              </option>
            ))}
          </Form.Control>
        </Col>
      </Row>
      <Row>
        <Col md={12} className="row-button">
          <Button
            variant="primary"
            size="sm"
            className="btn-add"
            onClick={() => history.push('/dashboard/delivery_reports/new')}
          >
            <Plus size={20} />
            <span>Thêm mới</span>
          </Button>
          <Button variant="success" size="sm" className="btn-import">
            <FilePlus size={20} />
            <span>Nhập file</span>
          </Button>
        </Col>
      </Row>

      <TablePaginationData
        columns={[
          {
            name: 'STT',
            field: 'index',
          },
          {
            name: 'Tên người mượn',
            field: 'user_name',
          },
          {
            name: 'Trạng thái',
            field: 'status',
          },
          {
            name: 'Ngày mượn',
            field: 'date_borrowed',
          },
          {
            name: 'Ngày trả',
            field: 'date_return',
          },
          {
            name: 'Tổng số lượng',
            field: 'total_amount',
          },
          {
            name: 'Đính kèm file',
            field: 'report_file',
          },
        ]}
        data={restructureData}
        loading={loadingItems}
      />
    </Wrapper>
  );
};

export default ListPage;
