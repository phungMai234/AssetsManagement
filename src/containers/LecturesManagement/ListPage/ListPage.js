import React, { useState, useMemo } from 'react';
import { Button, Row, Col } from 'react-bootstrap';

import BaseModal from 'components/BaseModal/BaseModal';
import Wrapper from './ListPage.styles';
import TablePaginationData from 'components/TablePaginationData';
import NewPage from '../NewPage';
import EditPage from '../EditPage';
import { Plus, Edit3, Trash2, AlertTriangle, FilePlus } from 'react-feather';
import { useQuery } from 'hooks/useQuery';
import useDelete from 'hooks/useDelete';
import ModalImportFile from '../ModalImportFile';
import BoxSearch from 'components/BoxSearch/BoxSearch';
import { filter, includes, lowerCase } from 'lodash';
import { useHistory } from 'react-router-dom';

const ListPage = () => {
  const [recordSelected, setRecordSelected] = useState();
  const [recordSelectedDel, setRecordSelectedDel] = useState();
  const [openModalImport, setOpenModalImport] = useState(false);
  const [params, setParams] = useState({});
  const history = useHistory();

  const { data: dataCategories, loading, force } = useQuery({ url: 'lecturers' });

  const [remove] = useDelete({
    id: recordSelectedDel?.id,
    nameCollection: 'lecturers',
    callback: () => {
      force();
    },
  });
  const recordItems = useMemo(() => {
    const customData = params.keyword
      ? filter(dataCategories, (item) => includes(lowerCase(item.name), lowerCase(params.keyword)))
      : dataCategories;
    return customData;
  }, [params, dataCategories]);

  const restructureData = useMemo(() => {
    if (!recordItems || !recordItems.length) return [];
    return recordItems.map((record, index) => ({
      ...record,
      index: index + 1,
      preview: (
        <Button
          variant="info"
          size="sm"
          onClick={() => {
            history.push(`/dashboard/lectures/${record.id}/detail?name=${record.name}`);
          }}
        >
          Chi tiết
        </Button>
      ),
      edit_row: (
        <Button
          variant="secondary"
          size="sm"
          onClick={() => {
            setRecordSelected(record);
          }}
        >
          Chỉnh sửa
        </Button>
      ),
      note: <div title={record?.note}>{record?.note}</div>,
      delete_row: (
        <Button
          variant="danger"
          size="sm"
          onClick={() => {
            setRecordSelectedDel(record);
          }}
        >
          <Trash2 size={20} />
        </Button>
      ),
    }));
  }, [history, recordItems]);

  const [modalAdd, setModalAdd] = useState(false);

  const toggleModal = () => {
    setModalAdd(!modalAdd);
  };

  return (
    <Wrapper>
      <Row>
        <Col md={4}>
          <BoxSearch
            placeholderText="Tìm kiếm theo tên"
            value={params.keyword}
            onChange={(e) => setParams({ ...params, keyword: e.target.value })}
          />
        </Col>
        <Col md={4}>
          <div className="header-content">
            <Button variant="success" size="sm" onClick={toggleModal}>
              <Plus size={20} />
              <span>Thêm mới</span>
            </Button>

            <Button variant="warning" size="sm" className="btn-import" onClick={() => setOpenModalImport(true)}>
              <FilePlus size={20} />
              <span>Nhập file</span>
            </Button>
          </div>
        </Col>
      </Row>

      <TablePaginationData
        columns={[
          {
            name: 'STT',
            field: 'index',
          },
          {
            name: 'Tên',
            field: 'name',
          },
          {
            name: 'Email',
            field: 'email',
          },
          {
            name: '',
            field: 'preview',
          },
          {
            name: '',
            field: 'edit_row',
          },
          {
            name: '',
            field: 'delete_row',
          },
        ]}
        data={restructureData}
        loading={loading}
      />
      <BaseModal
        show={!!recordSelectedDel}
        onConfirm={() => {
          remove();
          setRecordSelectedDel(null);
        }}
        onCancel={() => setRecordSelectedDel(null)}
        typeBtnConfirm="danger"
        confirmText="Xóa"
        typeModal="sm"
        content={
          <>
            <span
              style={{
                paddingRight: '5px',
              }}
            >
              <AlertTriangle size={50} fill={'#dc3545'} stroke="#fff" />
            </span>
            <span>Bạn có chắc chắn muốn xóa?</span>
          </>
        }
      />
      {modalAdd && <NewPage onCancel={() => setModalAdd(false)} />}
      {openModalImport && <ModalImportFile onCancel={() => setOpenModalImport(false)} />}
      {recordSelected && <EditPage record={recordSelected} onCancel={() => setRecordSelected(null)} />}
    </Wrapper>
  );
};

export default ListPage;
