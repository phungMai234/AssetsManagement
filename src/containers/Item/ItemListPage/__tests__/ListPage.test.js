import React from 'react';

import { render, findByText } from '@testing-library/react';

import ListPage from '../ItemList';
import { useQuery } from 'hooks/useQuery';
import { useQueryAssets } from 'hooks/useQueryAssets';

jest.mock('hooks/useQuery');
jest.mock('hooks/useQueryAssets');

describe('Item Edit Page', () => {
  beforeEach(() => {
    useQuery.mockReturnValue({
      loadingCates: false,
      dataCate: {
        data: [
          {
            id: '1',
            name: 'test',
            note: '',
            manager: 'Khoa công nghệ thông tin ',
          },
        ],
      },
    });
    useQueryAssets.mockReturnValue({
      loadingItems: false,
      dataItems: {
        data: [
          {
            name: 'Bàn phím Logitech K380',
            model_number: 'BPK380',
            serial_number: 'BPK380-003',
            id_category: 'fUACg168JQUzgSVI9DD3',
            status: 'Đang sử dụng',
            unit: 'cái',
            price_each: 560000,
            current_status: 'Tốt',
            purchase_date: {
              seconds: 1622030400,
              nanoseconds: 0,
            },
            image_detail: [
              {
                preview:
                  'https://firebasestorage.googleapis.com/v0/b/assets…=media&token=7e31dc76-25ee-4a02-a027-5db0470bd09e',
                path: 'bp3.png',
              },
            ],
            description: ' Bàn phím Logitech K380 không dây-Kết nối cùng lúc 3 thiết bị',
          },
        ],
      },
    });
  });

  it('should render correctly', () => {
    const container = render(<ListPage />);
    expect(container.container).toMatchSnapshot();
  });

  it('should render correctly with loading', () => {
    useQueryAssets.mockReturnValue({
      loadingItems: true,
      dataItems: {},
    });
    const wrapper = render(<ListPage />);
    const textLoading = wrapper.findByText('Loading...');
    expect(textLoading).toBeTruthy();
    expect(wrapper.container).toMatchSnapshot();
  });
});
