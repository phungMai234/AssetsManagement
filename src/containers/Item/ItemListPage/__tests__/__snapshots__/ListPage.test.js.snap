// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item Edit Page should render correctly 1`] = `
<div>
  <div
    class="sc-bdfBwQ kGpjlW"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-3 col-md-4"
      >
        <div
          class="sc-hKgILt kDhRMo"
        >
          <div
            class="mb-3 input-group"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
              >
                <svg
                  fill="none"
                  height="20"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <line
                    x1="21"
                    x2="16.65"
                    y1="21"
                    y2="16.65"
                  />
                </svg>
              </span>
            </div>
            <input
              class="form-control"
              placeholder="Tìm kiếm"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="col-lg-3 col-md-4"
      >
        <div
          class="sc-eCssSg kqYHFF"
        >
          <div
            class="icon-calendar"
          >
            <svg
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="18"
                rx="2"
                ry="2"
                width="18"
                x="3"
                y="4"
              />
              <line
                x1="16"
                x2="16"
                y1="2"
                y2="6"
              />
              <line
                x1="8"
                x2="8"
                y1="2"
                y2="6"
              />
              <line
                x1="3"
                x2="21"
                y1="10"
                y2="10"
              />
            </svg>
          </div>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class="date-picker-input"
                placeholder="Ngày mua"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-lg-3 col-md-4"
      >
        <select
          class="form-control"
        >
          <option
            value=""
          >
            Loại tài sản
          </option>
        </select>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="row-button col-md-12"
      >
        <button
          class="btn-add btn btn-success btn-sm"
          type="button"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="5"
              y2="19"
            />
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
          </svg>
          <span>
            Thêm mới
          </span>
        </button>
        <button
          class="btn-import btn btn-warning btn-sm"
          type="button"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8z"
            />
            <polyline
              points="14 2 14 8 20 8"
            />
            <line
              x1="12"
              x2="12"
              y1="18"
              y2="12"
            />
            <line
              x1="9"
              x2="15"
              y1="15"
              y2="15"
            />
          </svg>
          <span>
            Nhập file
          </span>
        </button>
        <button
          class="btn-add btn btn-info btn-sm"
          type="button"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
            />
            <polyline
              points="7 10 12 15 17 10"
            />
            <line
              x1="12"
              x2="12"
              y1="15"
              y2="3"
            />
          </svg>
          <span>
            Tải Danh sách
          </span>
        </button>
      </div>
    </div>
    <div
      class="sc-gsTCUz gTjdNH"
    >
      <div
        class="wrapper-select"
      >
        <label
          for="option"
        >
          Đang hiển thị: 
        </label>
        <select
          id="option"
        >
          <option
            value="5"
          >
            5
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="15"
          >
            15
          </option>
          <option
            value="20"
          >
            20
          </option>
        </select>
      </div>
      <div>
        <table
          class="table table-bordered table-hover"
        >
          <thead>
            <tr>
              <th
                class="index"
              >
                STT
              </th>
              <th
                class="picture"
              >
                Ảnh minh họa
              </th>
              <th
                class="model_number"
              >
                Model
              </th>
              <th
                class="serial_number"
              >
                Số seri (S/N)
              </th>
              <th
                class="name"
              >
                Tên
              </th>
              <th
                class="purchase_date"
              >
                Ngày mua
              </th>
              <th
                class="status"
              >
                Tình trạng sử dụng
              </th>
              <th
                class="price_each"
              >
                Giá (vnđ)
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="no-result"
                colspan="8"
              >
                Không có kết quả để hiển thị
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="wrapper-option"
      >
        <ul
          class="pagination"
        >
          <li
            class="page-item disabled"
          >
            <a
              aria-label="Go to first page"
              class="page-link"
              href="#"
            >
              «
            </a>
          </li>
          <li
            class="page-item disabled"
          >
            <a
              aria-label="Go to previous page"
              class="page-link"
              href="#"
            >
              ⟨
            </a>
          </li>
          <li
            class="page-item disabled"
          >
            <a
              aria-label="Go to next page"
              class="page-link"
              href="#"
            >
              ⟩
            </a>
          </li>
          <li
            class="page-item disabled"
          >
            <a
              aria-label="Go to last page"
              class="page-link"
              href="#"
            >
              »
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;

exports[`Item Edit Page should render correctly with loading 1`] = `
<div>
  <div
    class="sc-bdfBwQ kGpjlW"
  >
    <div
      class="row"
    >
      <div
        class="col-lg-3 col-md-4"
      >
        <div
          class="sc-hKgILt kDhRMo"
        >
          <div
            class="mb-3 input-group"
          >
            <div
              class="input-group-prepend"
            >
              <span
                class="input-group-text"
              >
                <svg
                  fill="none"
                  height="20"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="20"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="11"
                    cy="11"
                    r="8"
                  />
                  <line
                    x1="21"
                    x2="16.65"
                    y1="21"
                    y2="16.65"
                  />
                </svg>
              </span>
            </div>
            <input
              class="form-control"
              placeholder="Tìm kiếm"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="col-lg-3 col-md-4"
      >
        <div
          class="sc-eCssSg kqYHFF"
        >
          <div
            class="icon-calendar"
          >
            <svg
              fill="none"
              height="20"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <rect
                height="18"
                rx="2"
                ry="2"
                width="18"
                x="3"
                y="4"
              />
              <line
                x1="16"
                x2="16"
                y1="2"
                y2="6"
              />
              <line
                x1="8"
                x2="8"
                y1="2"
                y2="6"
              />
              <line
                x1="3"
                x2="21"
                y1="10"
                y2="10"
              />
            </svg>
          </div>
          <div
            class="react-datepicker-wrapper"
          >
            <div
              class="react-datepicker__input-container"
            >
              <input
                class="date-picker-input"
                placeholder="Ngày mua"
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="col-lg-3 col-md-4"
      >
        <select
          class="form-control"
        >
          <option
            value=""
          >
            Loại tài sản
          </option>
        </select>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="row-button col-md-12"
      >
        <button
          class="btn-add btn btn-success btn-sm"
          type="button"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <line
              x1="12"
              x2="12"
              y1="5"
              y2="19"
            />
            <line
              x1="5"
              x2="19"
              y1="12"
              y2="12"
            />
          </svg>
          <span>
            Thêm mới
          </span>
        </button>
        <button
          class="btn-import btn btn-warning btn-sm"
          type="button"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M14 2H6a2 2 0 0 0-2 2v16a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2V8z"
            />
            <polyline
              points="14 2 14 8 20 8"
            />
            <line
              x1="12"
              x2="12"
              y1="18"
              y2="12"
            />
            <line
              x1="9"
              x2="15"
              y1="15"
              y2="15"
            />
          </svg>
          <span>
            Nhập file
          </span>
        </button>
        <button
          class="btn-add btn btn-info btn-sm"
          type="button"
        >
          <svg
            fill="none"
            height="20"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
            />
            <polyline
              points="7 10 12 15 17 10"
            />
            <line
              x1="12"
              x2="12"
              y1="15"
              y2="3"
            />
          </svg>
          <span>
            Tải Danh sách
          </span>
        </button>
      </div>
    </div>
    <div
      class="sc-gsTCUz gTjdNH"
    >
      <div
        class="wrapper-select"
      >
        <label
          for="option"
        >
          Đang hiển thị: 
        </label>
        <select
          id="option"
        >
          <option
            value="5"
          >
            5
          </option>
          <option
            value="10"
          >
            10
          </option>
          <option
            value="15"
          >
            15
          </option>
          <option
            value="20"
          >
            20
          </option>
        </select>
      </div>
      <div>
        <table
          class="table table-bordered table-hover"
        >
          <thead>
            <tr>
              <th
                class="index"
              >
                STT
              </th>
              <th
                class="picture"
              >
                Ảnh minh họa
              </th>
              <th
                class="model_number"
              >
                Model
              </th>
              <th
                class="serial_number"
              >
                Số seri (S/N)
              </th>
              <th
                class="name"
              >
                Tên
              </th>
              <th
                class="purchase_date"
              >
                Ngày mua
              </th>
              <th
                class="status"
              >
                Tình trạng sử dụng
              </th>
              <th
                class="price_each"
              >
                Giá (vnđ)
              </th>
            </tr>
          </thead>
          <tbody>
            <tr>
              <td
                class="no-result"
                colspan="8"
              >
                Không có kết quả để hiển thị
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="wrapper-option"
      >
        <ul
          class="pagination"
        >
          <li
            class="page-item disabled"
          >
            <a
              aria-label="Go to first page"
              class="page-link"
              href="#"
            >
              «
            </a>
          </li>
          <li
            class="page-item disabled"
          >
            <a
              aria-label="Go to previous page"
              class="page-link"
              href="#"
            >
              ⟨
            </a>
          </li>
          <li
            class="page-item disabled"
          >
            <a
              aria-label="Go to next page"
              class="page-link"
              href="#"
            >
              ⟩
            </a>
          </li>
          <li
            class="page-item disabled"
          >
            <a
              aria-label="Go to last page"
              class="page-link"
              href="#"
            >
              »
            </a>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
